ThunderCloud/TSCAppDelegate.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  TSCAppDelegate.swift
+//  ThunderCloud
+//
+//  Created by Simon Mitchell on 24/08/2017.
+//  Copyright © 2017 threesidedcube. All rights reserved.
+//
+
+import UIKit
+import UserNotifications
+import ThunderRequest
+
+@UIApplicationMain
+/// A root app delegate which sets up your window and push notifications e.t.c.
+open class TSCAppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterDelegate {
```

Comment:
For `UNUserNotificationCenterDelegate`, the functions here that conform to that delegate could perhaps be moved to an extension of `TSCAppDelegate` that conforms specifically to that delegate. Will help with self-documenting the code too! :)

Reply:
Do you know whether this works with subclassing? I know in Objective-C it was hit-and-miss as to which methods would be called if you then subclassed `TSCAppDelegate` and wanted to override any of the delegate methods, but unsure about swift...

Reply:
From my thinking, if you create an extension of `TSCAppDelegate` (so not necessarily subclassing it), it should still call everything correctly. My recommendation would be to give it a shot to see if it does work :)